### ISSUE TITLE
__numpy_ufunc__ confusion in docs

### ISSUE THREAD

**Issue by jbrockmendel** [numpy/numpy#9252]
The top results for Google searches like "numpy subclass ndarray" point toward the [1.12.0](https://docs.scipy.org/doc/numpy-1.12.0/reference/arrays.classes.html) docs.  Three requests/suggestions.

1) A prominent "There is a newer version" link to the 1.13.0 docs.

2) The 1.12.0 page linked above directs the user to use `__numpy_ufunc__` for subclassing.  I'm not entirely clear on the timeline in #8247, but I couldn't make numpy_ufunc work with 1.12.0, so it isn't obvious whether the doc is accurate or (much more likely) I'm screwing up.

3) A non-toy example for numpy_ufunc and/or array_ufunc, ideally demonstrating the tie-ins or use cases for the array_prepare/array_finalize/...

**Comment by mhvk** [comment]
@jbrockmendel - not sure we can still change the 1.12.0 documentation, though if it is possible, I'd be all for it. In any case, only `__array_ufunc__` actually exists, and the 1.13.0 documentation has more information [1]. If you want a more complete/complicated example of its use, see https://github.com/astropy/astropy/pull/2583 for work I did for astropy's `Quantity` class.

[1] https://docs.scipy.org/doc/numpy/neps/ufunc-overrides.html, https://docs.scipy.org/doc/numpy/reference/arrays.classes.html

**Comment by jbrockmendel** [comment]
Thanks, I'd been using an older version of astropy.units as a template for poking at `__array_wrap__`, hadn't seen the version in 2583.  Using `__array_ufunc__` tentatively seems to solve my use case quite cleanly.

I had seen the neps/ufunc-overrides.html page, but did not see anything saying "this is now implemented".  Maybe that can go on the wish-list along with updating the 1.12.0 docs.

**Comment by mhvk** [comment]
Yes, someone elsewhere mentioned that the release notes seem to be missing! (It was in the e-mail @charris sent, but someone not elsewhere)

**Comment by rgommers** [comment]
I don't think we still want to amend the 1.12.0 docs at this point, however a ``There is a newer version`` banner for all new doc builds would be nice to have.

> Yes, someone elsewhere mentioned that the release notes seem to be missing!

That was fixed: https://docs.scipy.org/doc/numpy/release.html

> I had seen the neps/ufunc-overrides.html page, but did not see anything saying "this is now implemented".

https://docs.scipy.org/doc/numpy/neps/index.html has two sections, ``Implemented NEPS`` and ``Other NEPS``. https://docs.scipy.org/doc/numpy/neps/ufunc-overrides.html is under the former, and is up to date.
